### PR TITLE
Temporary fix for pug

### LINKED ISSUE #10: Pug component language prevents new auto a-la-carte from working
Use of pug in components seems to prevent the new auto a-la-carte from working as intended.

Here is an example repo: [https://github.com/shongololo/vuetify-a-la-carte](https://github.com/shongololo/vuetify-a-la-carte)

Example errors:

```
[Vue warn]: Unknown custom element: <v-app> - did you register the component correctly? For recursive components, make sure to provide the "name" option.

found in

---> <App> at src/components/App.vue
       <Root>
```

### PR DESCRIPTION
**Pull request by amritk**
Fixes #10

### CODE REVIEW

File: lib/loader.js [hunk 1]
```diff
@@ -22,11 +23,17 @@ module.exports = function (content, sourceMap) {
...
     const component = compiler.parseComponent(file)
-    component.template && compiler.compile(component.template.content, {
-      modules: [{
-        postTransformNode: node => { tags.add(node.tag) }
-      }]
-    })
+    if (component.template) {
+      // Temp fix for pug
+      if (component.template.lang === 'pug') {
+        component.template.content = pug.compile(component.template.content)()
```
Comment (KaelWD) [inline lib/loader.js]: Move pug to `peerDependencies` and only require it in this block.

Comment (amritk) [reply lib/loader.js]: Done

File: lib/loader.js [hunk 2]
```diff
@@ -22,11 +22,17 @@ module.exports = function (content, sourceMap) {
     const tags = new Set()
     const file = this.fs.readFileSync(this.resourcePath).toString('utf8')
     const component = compiler.parseComponent(file)
-    component.template && compiler.compile(component.template.content, {
-      modules: [{
-        postTransformNode: node => { tags.add(node.tag) }
-      }]
-    })
+    if (component.template) {
+      const pug = require('pug')
```
Comment (KaelWD) [inline lib/loader.js]: In both ifs, it's only needed if pug is being used.

Comment (amritk) [reply lib/loader.js]: whoops

Comment (amritk) [reply lib/loader.js]: Done

**Comment by KaelWD** [comment]
Thanks!

**Comment by Coder-256** [comment]
@KaelWD Is it really necessary to add pug as a `peerDependency`? Now I get this warning when running `npm install`:

    npm WARN vuetify-loader@1.0.6 requires a peer of pug@^2.0.3 but none is installed. You must install peer dependencies yourself.

I am not using pug templates.

**Comment by KaelWD** [comment]
No

**Comment by Coder-256** [comment]
@KaelWD I meant, was it necessary for `vuetify-loader` to add pug as a peer dependency, because it is causing warnings in my project

**Comment by amritk** [comment]
I don't know too much about optional dependancies but here's an issue where they talk about it:
https://github.com/npm/npm/issues/3066
And this one https://github.com/npm/npm/issues/20646

> I propose that an "optionalPeerDependencies" field should behave like "peerDependencies", except that a warning should only be emitted if an incompatible version of a dependency is actually installed; missing dependencies should be ignored.

**Comment by Berkmann18** [comment]
Is there a way for `vuetify-loader` to make `pug` like an "optionalPeerDependencies"?
It's fustrating to see the npm errors on `npm ls` when `pug` isn't being used at all.

**Pull request merged**